design/events.md

```diff
@@ -0,0 +1,417 @@
+      <td><strong>Sub-category</strong></td>
+      <td><strong>Event type</strong></td>
+      <td><strong>Triggered by</strong></td>
+      <td><strong>Specific event attributes</strong></td>
+      <td><strong>Comments</strong></td>
+    </tr>
+    <tr>
+      <td rowspan="6">Git-like events</td>
+      <td rowspan="3">Reference events</td>
+      <td><code>REFERENCE_CREATED</code></td>
+      <td>API call</td>
+      <td>
+        <ul>
+          <li>Reference name</li>
+          <li>Reference type (Branch, Tag...)</li>
+          <li>New HEAD</li>
+        </ul>
+      </td>
+      <td>In all reference events, the reference name should be included in short form 
+          (e.g. <code>branch1</code>) and in full form (e.g. <code>/refs/heads/branch1</code>)</td>
+    </tr>
+    <tr>
+      <td><code>REFERENCE_UPDATED</code></td>
+      <td>API call</td>
+      <td>
+        <ul>
+          <li>Reference name</li>
+          <li>Reference type (Branch, Tag...)</li>
+          <li>Previous HEAD</li>
+          <li>New HEAD</li>
+        </ul>
+      </td>
+      <td>I.e., reassigned to a different hash, through an API call exclusively. Indirect reference updates due to a commit 
+          do not trigger this event.</td>
+    </tr>
+    <tr>
+      <td><code>REFERENCE_DELETED</code></td>
+      <td>API call</td>
+      <td>
+        <ul>
+          <li>Reference name</li>
+          <li>Reference type (Branch, Tag...)</li>
+          <li>Previous HEAD</li>
+        </ul>
+      </td>
+      <td>&nbsp;</td>
+    </tr>
+    <tr>
+      <td rowspan="3">Committing events</td>
+      <td><code>COMMIT</code></td>
+      <td>API call</td>
+      <td>
+        <ul>
+          <li>Branch name</li>
+          <li>Previous HEAD</li>
+          <li>New HEAD</li>
+          <li><code>CommitMeta</code> object</li>
+        </ul>
+      </td>
+      <td>Operations (PUTs and DELETEs) are not included, in order to avoid huge event payloads. 
+          Full operation contents are sent separately as content-related events.
+        <p>A <code>COMMIT</code> event should trigger:</p>
+        <ul>
+          <li>0-N content events (one for each PUT/DELETE operation).</li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td><code>MERGE</code></td>
+      <td>API call</td>
+      <td>
+        <ul>
+          <li>Source branch name</li>
+          <li>Source branch HEAD</li>
+          <li>Target branch name</li>
+          <li>Previous target branch HEAD</li>
+          <li>New target branch HEAD</li>
+          <li>Common ancestor hash</li>
+        </ul>
+      </td>
+      <td>Operations (PUTs and DELETEs) are not included, in order to avoid huge event payloads. 
+          Full operation contents are sent separately as content-related events.
+        <p>A <code>MERGE</code> event should trigger:</p>
+        <ul>
+          <li>0-N content events (one for each global PUT/DELETE operation, excluding intermediary operations).</li>
```

Comment:
```suggestion
          <li>0-N content events (one for each PUT/DELETE operation, excluding intermediary operations).</li>
```

Let's avoid the term "global" in conjunction with commits - it has some "bad taste" wrt "global state" ;)